frontend/test/metabase/lib/expressions/resolver.unit.spec.js

```diff
@@ -159,6 +159,44 @@ describe("metabase-lib/expressions/resolve", () => {
     it("should honor CONCAT's implicit casting", () => {
       expect(() => expr(["concat", ["coalesce", "B", 1]])).not.toThrow();
     });
+
+    describe("datetime commands", () => {
+      it("should resolve unchained commmands", () => {
+        expect(() => expr(["get-week", "2022-01-01"])).not.toThrow();
+        expect(() => expr(["get-week", A])).not.toThrow();
+
+        expect(() =>
+          expr(["datetime-add", "2022-01-01", 1, "month"]),
+        ).not.toThrow();
+      });
+
+      it("should resolve chained commmands", () => {
+        expect(() =>
+          expr([
+            "datetime-subtract",
+            ["datetime-add", "2022-01-01", 1, "month"],
+            2,
+            "minute",
+          ]),
+        ).not.toThrow();
+      });
+
+      it("should chain datetime commands onto commands of compatible types", () => {
+        expect(() =>
+          expr([
+            "concat",
+            ["datetime-add", "2022-01-01", 1, "month"],
+            "a string",
+          ]),
+        ).not.toThrow();
```

Comment:
Potentially off topic - why isn't the arg type of concat "expression", and not "string"?

Reply:
concat works with IDs, timestamps, as well a strings, and maybe other types.


<details>
  <summary>See examples in action</summary>
  
<img width="254" alt="image" src="https://user-images.githubusercontent.com/380816/204897822-7b45b82a-6682-41d1-afb0-fb8770d59fbb.png">

<img width="364" alt="image" src="https://user-images.githubusercontent.com/380816/204897834-e63433db-d222-4e43-b122-5b619c758782.png">
  ```
</details>

Reply:
cool. learnt something